lib/rules/no-invalid-this.js

```diff
@@ -140,8 +142,28 @@ module.exports = {
 				stack.pop();
 			},
 
+			"AccessorProperty > *.value"(node) {
+				stack.push({
+					init: true,
+					node,
+					valid: true,
+				});
+			},
+
+			"AccessorProperty:exit"() {
+				stack.pop();
+			},
```

Comment:
Can you also add a test to verify that this works as intended? Currently, if this code is removed, no tests fail.

Reply:
Added!
```ts
function foo() {
	class C {
	  accessor a = this.b;
	  accessor [this.c] = foo;
	}
}
```

rule won't report this case if we remove this part.